src/eventbridge/bus.ts

```diff
@@ -0,0 +1,114 @@
+import { aws_events } from "aws-cdk-lib";
+import { Construct } from "constructs";
+import { EventBusRule, EventPredicateFunction } from "./rule";
+import { EventBusRuleInput } from "./types";
+
+/**
+ * A Functionless wrapper for a AWS CDK {@link aws_events.EventBus}.
+ *
+ * Filtering events and sending them to Lambda.
+ *
+ * ```ts
+ * interface Payload {
+ *    value: string;
+ * }
+ *
+ * // An event with the payload
+ * type myEvent = EventBusRuleInput<Payload>;
+ *
+ * const myAwsFunction = new aws_lambda.Function(this, 'myFunction', { ... });
+ * // A function that expects the payload.
+ * const myLambdaFunction = new Function<Payload, void>();
```

Comment:
this should wrap `myAwsFunction`

Also, should you use `functionless.Function` so it's clear that it is imported?